### PR TITLE
`Notifications`: Fix broken push notifications for mobile apps

### PR DESCRIPTION
**Pull request by TimOrtel**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Recent changes to the notification component broke the push notifications for Android and iOS devices. In order to prepare for the upcoming release of the Android app, we should fix these asap so we can start testing. 

### Description

Readded conversation name to reply notification

### Steps for Testing
1. Install the Android App on your Android Emulator
2. Register for push notifications in the Android App
3. Send yourself some notifications from the web client

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
  - Added new constructors, fields, and methods to enhance conversation functionality.
- **Bug Fixes**
  - Improved data handling and processing for conversations and messages.
- **Refactor**
  - Updated the way `conversationName` is retrieved in `ConversationNotificationService`.
- **Documentation**
  - Added `copy()` method override in `Conversation`, `GroupChat`, and `OneToOneChat` classes.

### CODE REVIEW

**Review by AntonGeTUM — DISMISSED** [review]
Tested on ts11 and an Android emulator, works as expected
image

**Review by Strohgelaender — CHANGES_REQUESTED** [review]

**Review by Strohgelaender — DISMISSED** [review]
code lgtm, participants are still hidden when sending the notification but loaded while creating it.

**Review by nityanandaz — APPROVED** [review]
Thanks for fixing notifications

**Review by nityanandaz — APPROVED** [review]
Code review: re-approve; improved access to `getHumanReadableNameForReceiver`

**Review by Strohgelaender — APPROVED** [review]

**Review by basak-akan — APPROVED** [review]
Tested during the testing session on TS1, notifications worked well

**Review by nata-81 — DISMISSED** [review]
Tested during testing session works well.

**Review by Kroko-fant — APPROVED** [review]
Tested in a testing session

**Review by Jan-Thurner — DISMISSED** [review]
Tested in a testing session, worked well.

**Review by rabeatwork — APPROVED** [review]

File: src/main/java/de/tum/in/www1/artemis/repository/metis/conversation/ConversationRepository.java [hunk 1]
```diff
@@ -29,6 +30,9 @@ public interface ConversationRepository extends JpaRepository<Conversation, Long
     // This is used only for testing purposes
     List<Conversation> findAllByCourseId(long courseId);
 
+    @EntityGraph(type = LOAD, attributePaths = { "conversationParticipants" })
+    Optional<Conversation> findWithParticipantsById(long id);
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/repository/metis/conversation/ConversationRepository.java:34]:
```suggestion
    Optional<Conversation> findWithParticipantsById(long conversationId);
```

**Pull request merged**